x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AgentConfigurationCreateEdit/ServicePage/ServicePage.tsx

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import {
+  EuiTitle,
+  EuiSpacer,
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton
+} from '@elastic/eui';
+import React from 'react';
+import { i18n } from '@kbn/i18n';
+import { isString } from 'lodash';
+import { AgentConfigurationIntake } from '../../../../../../../../../../plugins/apm/common/runtime_types/agent_configuration/configuration_types';
+import {
+  omitAllOption,
+  getOptionLabel
+} from '../../../../../../../../../../plugins/apm/common/agent_configuration_constants';
+import { useFetcher } from '../../../../../../hooks/useFetcher';
+import { FormRowSelect } from './FormRowSelect';
+import { CancelButton } from './CancelButton';
+
+interface Props {
+  newConfig: AgentConfigurationIntake;
+  setNewConfig: React.Dispatch<React.SetStateAction<AgentConfigurationIntake>>;
+  onClickNext: () => void;
+}
+
+export function ServicePage({ newConfig, setNewConfig, onClickNext }: Props) {
+  const { data: serviceNames = [], status: serviceNamesStatus } = useFetcher(
+    callApmApi => {
+      return callApmApi({
+        pathname: '/api/apm/settings/agent-configuration/services',
+        forceCache: true
+      });
+    },
+    [],
+    { preservePreviousData: false }
```

Comment:
Just for curiosity, how the `forceCache` and `preservePreviousData` work?

Reply:
Only requests with time ranges are cached. So by default this will not be cached.

Reply:
`preservePreviousData` determines whether the previous response should be overwritten with `undefined` when a new response starts loading.

https://github.com/elastic/kibana/blob/6d1479fc08147b78454c6a5c1aee2b44d894dfd1/x-pack/legacy/plugins/apm/public/hooks/useFetcher.tsx#L69-L73